pkg/controller/common/annotation/timeout.go

```diff
@@ -0,0 +1,44 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package annotation
+
+import (
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// ExtractTimeout extracts a timeout value specified as an annotation on a resource.
+func ExtractTimeout(obj runtime.Object, annotation string, defaultVal time.Duration) time.Duration {
+	if obj == nil {
+		return defaultVal
+	}
+
+	metaAcc := meta.NewAccessor()
+
+	ann, err := metaAcc.Annotations(obj)
+	if err != nil {
+		log.V(1).Info("Failed to extract annotations from object", "error", err, "object", obj)
+		return defaultVal
+	}
+
+	if len(ann) == 0 {
```

Comment:
Just curious: what is the advantage of this explicit handling of an empty annotation map over just relying on the lookup?

Reply:
The main purpose is to check that the map is not nil. It has the added bonus of not requiring a lookup when the map is empty as well.

Reply:
I think lookups work even on `nil` maps afaik.

Reply:
Yes, they do. Interestingly, the nil check seems to make things faster -- which kind of makes sense because the length is a type attribute that can be checked efficiently.  It doesn't matter much in this context though.

```
BenchmarkNil/direct-8          	456756687	        2.56 ns/op	      0 B/op	      0 allocs/op
BenchmarkNil/nilcheck-8        	1000000000	        0.662 ns/op	      0 B/op	      0 allocs/op
```